OpenRA.Mods.Common/Traits/Wanders.cs

```diff
@@ -28,36 +28,47 @@ public abstract class WandersInfo : ITraitInfo
 		[Desc("Maximum amount of ticks the actor will sit idly before starting to wander.")]
 		public readonly int MaxMoveDelay = 0;
 
-		public abstract object Create(ActorInitializer init);
+		public override object Create(ActorInitializer init) { return new Wanders(init.Self, this); }
 	}
 
-	public class Wanders : INotifyIdle, INotifyBecomingIdle
+	public class Wanders : UpgradableTrait<WandersInfo>, INotifyCreated, INotifyIdle, INotifyBecomingIdle
 	{
 		readonly Actor self;
 		readonly WandersInfo info;
+		IResolveOrder move;
```

Comment:
This approach was suggested by @pchote to avoid having to explicitly look up both `Mobile` and `Aircraft`.